### ISSUE TITLE
[unstable] save fail to open at latest unstable version

### ISSUE THREAD

**Issue by 1stLtv2** [Regalis11/Barotrauma#12996]
### Disclaimers

- [x] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

multiplayer campaign created  at 15SEP2023 and runs in unstable version
the save last runs at 09NOV2023. The game fail to load the file after the latest hotfix(update) . All other non-unstable save can run in unstable, and that unstable save can still run in public version.
![barotrauma](https://github.com/Regalis11/Barotrauma/assets/65464699/63272300-688d-4034-96e9-01aa59dbce6d)

### Bug prevalence

Happens every now and then

### Single player or multiplayer?

Multiplayer hosted from the in-game menu (= using a listen server)

### Version

Other

### -

unstable v1.2.0.0

### Which operating system did you encounter this bug on?

Windows

**Comment by Regalis11** [comment]
Thank you for the report! Would you mind sending us the save? Note that you need to .zip it first because github doesn't allow attaching files with the .save file extension.

**Comment by 1stLtv2** [comment]
[11-8.zip](https://github.com/Regalis11/Barotrauma/files/13319226/11-8.zip)
There you are

**Comment by Rokvach** [comment]
Tested with the provided save, I was able to run it in Stable and on the fixed PR but not on Unstable so this is working correctly. 

Also tested server side error showing in lobby by trying to load up a singleplayer save in multiplayer and the error is displayed correctly on the chat. 
![image](https://github.com/Regalis11/Barotrauma/assets/7299826/6c56e7f2-c953-4f91-8eda-2bdf753bbc81)


Working correct, can be merged and closed.
